src/components/secrets/Secret.tsx

```diff
@@ -696,22 +972,55 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
             <div className="form__row">
                 <label className="form__label">Data type</label>
                 <div className="form-row__select-external-type flex">
-                    <Select
-                        value={externalType}
-                        onChange={(e) => {
-                            setExternalType(e.target.value)
-                        }}
-                        disabled={secretMode}
-                    >
-                        <Select.Button>
-                            {externalType || externalType === '' ? EXTERNAL_TYPES[externalType] : `Select Secret Type`}
-                        </Select.Button>
-                        {Object.entries(EXTERNAL_TYPES).map(([value, name]) => (
-                            <Select.Option key={value} value={value}>
-                                {name}
-                            </Select.Option>
-                        ))}
-                    </Select>
+                    <ReactSelect 
+                    placeholder="Select Secret Type"
+                    options={getTypeGroups()}
+                    defaultValue={externalType && externalType !== '' ? getTypeGroups(externalType) : getTypeGroups()[0].options[0]}
+                    onChange={onChange}
+                    styles={{
+                        ...multiSelectStyles,
+                        menu: (base) => ({ ...base, zIndex: 9999, textAlign: 'left' }),
+                        control: (base) => ({...base, border: '1px solid #d6dbdf', width: '450px'}),
+                        group: (base) => ({
+                            ...base,
+                            paddingTop: 0,
+                            paddingBottom: 0,
+                        }),
+                        groupHeading: (base) => ({
+                            ...base,
+                            fontWeight: 600,
+                            fontSize: '12px',
+                            height: '28px',
+                            color: 'var(--N900)',
+                            backgroundColor: 'var(--N100)',
+                            marginBottom: 0,
+                        }),
+                        indicatorsContainer: (provided, state) => ({
+                            ...provided,
+                        })
+                    }}
+                    components={{
+                        IndicatorSeparator: null,
+                        Option: (props) => {
+                            props.selectProps.styles.option = getCustomOptionSelectionStyle()
```

Comment:
same for the components. please extract out and create separate function.

Reply:
Created